activemq-jaas/src/main/java/org/apache/activemq/jaas/LDAPLoginModule.java

```diff
@@ -320,20 +316,19 @@ protected boolean authenticate(String username, String password) throws LoginExc
                 throw new FailedLoginException("Password does not match for user: " + username);
             }
         } catch (CommunicationException e) {
+            closeContext();
             FailedLoginException ex = new FailedLoginException("Error contacting LDAP");
             ex.initCause(e);
             throw ex;
         } catch (NamingException e) {
-            if (context != null) {
-                close(context);
-            }
+            closeContext();
             FailedLoginException ex = new FailedLoginException("Error contacting LDAP");
             ex.initCause(e);
             throw ex;
         }
 
         if (context != null) {
```

Comment:
You don't need to check if context != null as in closeContext method it's already checked

Reply:
fixed